cdap-sentry/cdap-sentry-extension/cdap-sentry-binding/src/main/java/co/cask/cdap/security/authorization/sentry/binding/AuthBinding.java

```diff
@@ -281,11 +279,19 @@ public Void run(SentryGenericServiceClient client) throws Exception {
       if (authorizables.isEmpty()) {
         continue;
       }
-      EntityId parent = null;
+      EntityType entityType = null;
+      Map<EntityType, String> entityParts = new LinkedHashMap<>();
       for (TAuthorizable authorizable : authorizables) {
-        parent = toEntityId(authorizable, parent);
+        entityType = toEntityId(authorizable, entityParts);
```

Comment:
Let's add more comments here. Saying we only keep the final entity type since the authorizable is ordered top down, etc.